hail/python/hail/expr/expressions/base_expression.py

```diff
@@ -761,8 +766,13 @@ def collect(self):
         :obj:`list`
         """
         uid = Env.get_uid()
-        t = self._to_table(uid).key_by()
-        return [r[uid] for r in t._select("collect", hl.struct(**{uid: t[uid]})).collect()]
+        t = self._to_table(uid).key_by().select(uid)
+
+        e = t.collect(_localize=False).map(lambda r: r[uid])
```

Comment:
I'm having trouble deciphering this code. Why was `_select` needed before, but now you have a plain select?

Reply:
I don't think it was needed before.